app/scripts/services/bindingModalUtils.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('BindingModalUtils', [
+    function() {
+      return {
+        sortApplications: function(deploymentConfigs,
+                                    deployments,
+                                    replicationControllers,
+                                    replicaSets,
+                                    statefulSets) {
+          // Don't waste time sorting on each data load, just sort when we have them all
+          if (deploymentConfigs && deployments && replicationControllers && replicaSets && statefulSets) {
+            var apiObjects = deploymentConfigs.concat(deployments)
+                                              .concat(replicationControllers)
+                                              .concat(replicaSets)
+                                              .concat(statefulSets);
```

Comment:
I think it's more than this... If you pass an object as an argument to concat, does it just append the object?

Maybe a moot point if you're rolling back the retractor though